### PR TITLE
fix: Apply `paren-remover` while minifying

### PR DESCRIPTION
**Pull request by kdy1**
### Description

This is required to fix https://vercel.slack.com/archives/C04KC8A53T7/p1717599802994489

### Testing Instructions

Closes SWC-171

### CODE REVIEW

**Review by timneutkens — DISMISSED** [review]
The PR seems to be missing a test to prevent future regression 👀

File: crates/turbopack-ecmascript/src/minify.rs [hunk 1]
```diff
@@ -63,6 +64,8 @@ pub async fn minify(path: Vc<FileSystemPath>, code: Vc<Code>) -> Result<Vc<Code>
             let top_level_mark = Mark::new();
 
             Ok(compiler.run_transform(handler, false, || {
+                let program = program.fold_with(&mut paren_remover(Some(&comments)));
```
Comment (eps1lon) [inline crates/turbopack-ecmascript/src/minify.rs:67]: Can the minifier check if that fixer was applied?

Considering "Fixes ast nodes before printing so semantics are preserved.", it feels like   somewhere we should crash if the fixer isn't applied. 

In other words: When would you ever want to not preserve semantics?

**Review by sokra — APPROVED** [review]

File: crates/turbopack-ecmascript/src/minify.rs (hunk 1 above)
Comment (kdy1) [reply crates/turbopack-ecmascript/src/minify.rs:67]: It should be done by `codegen`, but we can't know if it's applied from codegen

File: crates/turbopack-tests/tests/snapshot/minification/paren-remover/input/index.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
+function toFixed(value, maxDecimals, roundingFunction, optionals) {
```
Comment (kdy1) [inline crates/turbopack-tests/tests/snapshot/minification/paren-remover/input/index.js:1]: This test is copied from https://github.com/swc-project/swc/pull/8442

Comment (kdy1) [reply crates/turbopack-tests/tests/snapshot/minification/paren-remover/input/index.js:1]: Note: I also verified this fix by disabling `paren_remover` in the testing system for SWC minifier

**Pull request merged**